core/shared/src/main/scala/spire/math/Interval.scala

```diff
@@ -767,6 +767,48 @@ sealed abstract class Interval[A](implicit order: Order[A]) extends Serializable
 
   def foldOver[B](init: B, step: A)(f: (B, A) => B)(implicit ev: AdditiveMonoid[A], nt: NumberTag[A]): B =
     iterator(step).foldLeft(init)(f)
+
+  /**
+    * Result of overlapping this interval with another one.
+    * Can be one of the following:
+    * - [[spire.math.interval.Equals]] if intervals are equal
+    * - [[spire.math.interval.StrictlyLess]] if intervals are notEmpty don't intersect
+    * - [[spire.math.interval.LessAndOverlaps]] if intervals intersect and neither is a subset of another
+    * - [[spire.math.interval.Subset]] if one interval (possibly empty) is a subset of another
+    *
+    * Note that lhs and rhs can be switched places in any result, except [[spire.math.interval.Equals]].
+    * For example (pseudo-code):
+    * {
+    * val lhs = [5, 6]
+    * val rhs = (0, 1)
+    *
+    * // this returns StrictlyLess(rhs, lhs). Note that lhs and rhs switched places in compare to the overlap call
+    * lhs.overlap(rhs)
+    * }
+    */
+  def overlap(rhs: Interval[A]): Overlap[A] = {
```

Comment:
We can use this to define a `PartialOrder` on `Interval` (where <= is defined for `StrictlyLess` and `Equals`) - may be nice to add that instance here.